src/Compilers/Shared/ShadowCopyAnalyzerAssemblyLoader.cs

```diff
@@ -41,12 +44,22 @@ public ShadowCopyAnalyzerAssemblyLoader(string baseDirectory = null)
                 _baseDirectory = Path.Combine(Path.GetTempPath(), "CodeAnalysis", "AnalyzerShadowCopies");
             }
 
-            Task.Run((Action)DeleteLeftoverDirectories);
+            DeleteLeftoverDirectoriesTask = Task.Run((Action)DeleteLeftoverDirectories);
         }
 
         private void DeleteLeftoverDirectories()
         {
-            foreach (var subDirectory in Directory.EnumerateDirectories(_baseDirectory))
+            IEnumerable<string> subDirectories;
+            try
+            {
+                subDirectories = Directory.EnumerateDirectories(_baseDirectory);
```

Comment:
My assumption was that other exceptions (say, `UnauthorizedAccessException`) would also be thrown when `_baseDirectory` is used later (when shadow copying), so there was no advantage in ignoring those exceptions here.